R/plot_section.R

```diff
@@ -0,0 +1,207 @@
+#' Generate stratigraphic section plot
+#'
+#' A function to plot the stratigraphic ranges of fossil taxa from occurrence
+#' data.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences containing
+#'   at least two columns: names of taxa, and their stratigraphic position (see
+#'   `name` and `level` arguments).
+#' @param name \code{character}. The name of the column you wish to be treated
+#'   as the input names, e.g. "taxon" (default).
+#' @param level \code{character}. The name of the column you wish to be treated
+#'   as the stratigraphic levels associated with each occurrence, e.g. "bed"
+#'   (default) or "height".
+#' @param certainty \code{character}. The name of the column you wish to be
+#'   treated as the information on whether an identification is certain ("1") or
+#'   uncertain ("0"). By default (\code{certainty = NULL}), no column name is
+#'   provided, and all occurrences are assumed to be certain. In the plot, any
+#'   occurrence labelled as certain will be plotted with a black circle, while
+#'   any occurrence labelled as uncertain will be plotted with a white circle.
+#' @param by \code{character}. How should the output be sorted? Either: "FAD"
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param xlab \code{character}. The x-axis label (over taxa) passed directly to
+#'   \code{plot}.
+#' @param ylab \code{character}. The y-axis label (over strata) passed directly
+#'   to \code{plot}.
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()].
+#' @param x_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis.
+#' @param y_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis.
+#'
+#' @return Invisibly returns a data.frame of the calculated taxonomic
+#'   stratigraphic ranges.
+#'
+#'   Function is usually used for its side effect, which is to create a plot
+#'   showing the stratigraphic ranges of taxa in a section, with levels at which
+#'   the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s): Bethany Allen & Alexander Dunhill
+#' @section Reviewer(s): William Gearty & Lewis A. Jones
+#' @importFrom graphics axis par segments plot points box
+#'
+#' @examples
```

Comment:
I thought it might be better to showcase this in one of the vignettes, if possible?

Reply:
We could do that too, but I suspect a lot of people don't go to the vignettes. If we provide something simple in the examples, I think this would be most useful. Are you concerned about the length of the examples?

Reply:
Yeah, I was mostly worried about the length of the examples